engine/lib/comments.php

```diff
@@ -275,6 +275,10 @@ function _elgg_comments_permissions_override($hook, $type, $return, $params) {
  * @return array $returnvalue Modified mail parameters
  */
 function _elgg_comments_notification_email_subject($hook, $type, $returnvalue, $params) {
+	if (!is_array($returnvalue)) {
+		// another hook handler returned a non-array, let's not override it
+		return;
```

Comment:
IMHO, should still return the $returnvalue to the pipeline, in case someone wants to do something with it